src/main/java/org/candlepin/model/EnvironmentCurator.java

```diff
@@ -71,19 +69,88 @@ public List<Consumer> getEnvironmentConsumers(String environmentId) {
             .getResultList();
     }
 
-    public CandlepinQuery<Environment> listForOwner(Owner o) {
-        DetachedCriteria criteria = this.createSecureDetachedCriteria()
-            .add(Restrictions.eq("owner", o));
+    public List<Environment> listForOwnerByType(Owner o, List<String> type) {
+        String jpql = "SELECT e FROM Environment e WHERE e.owner.id = :owner_id and e.type in :type";
+        return this.getEntityManager()
+            .createQuery(jpql, Environment.class)
+            .setParameter("owner_id", o.getId())
+            .setParameter("type", type.stream().map(x -> x.toLowerCase()).toList())
+            .getResultList();
+    }
+
+    public List<Environment> listForOwnerIgnoreType(Owner o) {
+        String jpql = "SELECT e FROM Environment e WHERE e.owner.id = :owner_id";
+        return this.getEntityManager()
+            .createQuery(jpql, Environment.class)
+            .setParameter("owner_id", o.getId())
+            .getResultList();
+    }
+
+    public List<Environment> listForOwnerNullType(Owner o) {
+        String jpql = "SELECT e FROM Environment e WHERE e.owner.id = :owner_id and e.type is null";
+        return this.getEntityManager()
+            .createQuery(jpql, Environment.class)
+            .setParameter("owner_id", o.getId())
+            .getResultList();
+    }
 
-        return this.cpQueryFactory.buildQuery(this.currentSession(), criteria);
+    public List<Environment> listForOwnerByNameAndType(Owner o, String envName, List<String> type) {
```

Comment:
nit: For the methods where the overall behavior isn't changing but the criteria is, we can just use one method name and overload it to avoid the Java naming stereotype in some of these cases.